### ISSUE TITLE
Possible to add NL subscribers from the Admin Panel using Store View specific settings?

### ISSUE THREAD

**Issue by mobweb** [ebizmarts/magemonkey#61]
We have the following setup:

- Website A
-- Store A1
--- Store View AA1

- Website B
-- Store B1
--- Store View BB1

I have disabled MageMonkey in the *Website A* and global configuration. I have enabled it in the *Store A1* configuration and have entered my API credentials in the *Store View AA1* configuration.

Now, when I go to *Customers -> Manage Customers* in the Admin Panel and add a customer that is assigned to *Store View AA1* to the newsletter list, I can see them listed under *Newsletter -> Newsletter Subscribers*. But MageMonkey doesn't do anything. There are no log-entries, and no subscribers are added in my Mailchimp account.

Should it be possible to add subscribers using the technique I just added above? We really need to be able to link each Store View or website to a separate Mailchimp account, as each website is run by a separate company with different Mailchimp accounts for billing reasons.

**Comment by centerax** [comment]
Thanks for your feedback.
We will revie this and get you an answer as soon as possible.

**Comment by mobweb** [comment]
Thanks for the quick feedback. I noticed that if I switch to the *Store View AA1* configuration in the Admin Panel and save the MageMonkey settings for that specific Store View, I get a few log entries, the last one of them being:

	2015-03-31T19:01:12+00:00 DEBUG (7): Array
	(
	    [0] => Error: API Key can not be blank, code 104
	    [__pid] => 5870
	)

So it looks like it's still maybe using some of the default values?

Do let me know if you need any more information or if you want me to set you up with access to our testing environment where you can reproduce the error.

**Comment by Santiagoebizmarts** [comment]
@mobweb I'm sorry for the delay on this. Is this happening on latest MageMonkey version (1.1.28)?
This seems to be working okay for me when I test it locally.

**Comment by mobweb** [comment]
(Sorry, I haven't had a chance to look at this issue yet, but it's one of my next tasks on my list. I will update you as soon as I have a chance to look at this again.)

**Comment by centerax** [comment]
Thanks @mobweb Please feel free to reopen if you need to.

Thanks again.

**Comment by mobweb** [comment]
I have just updated to the latest version, and it still doesn't seem to work. Let me describe our current setup:

- One Magento installation with two websites (Website A, Website B) and one Store View per website
- Two Mailchimp accounts: Account A, Account B, with one Suscriber List each

In our Magento configuration, I have set the "Default Configuration" to use Account A. Then I have switched to the configuration scope of the Store View of Website B. There I have updated the "API Key" field to use the key of Account B. I have also untoggled "Use Default" for the "Account Details" and "General Subscription" values. In the "General Subscription" dropdown I was able to select the Subscriber List of Account B successfully.

Now when I go to the frontend of Website A and subscribe to the newsletter, the subscriber shows up in the Subscriber List for Account A, just as expected. When I switch to Website B in the frontend and also entere a test address to be subscribed to the Newsletter, it correctly shows up in the Subscriber List for Account B, just as expected.

So far, everything has worked as expected.

Now when I go to the "Manage Customers" page in the Magento Admin Panel and select a customer that is assigned to Website A, I can choose the "Subscribe to Newsletter" action and submit that action. A few minutes later, that subscriber has correctly been added to the Subscriber List in Account A in Mailchimp, as well as the "Newsletter Subscribers" list in Magento itself.

However, when I select a customer that is assigned to Website B and submit the "Subscribe to Newsletter" action, nothing happens in Mailchimp. In Magento, the newsletter susbcriber list is updated correctly.

Looking at the log, I can see the following entry at the bottom:

	2015-04-15T09:36:38+00:00 DEBUG (7): Array
	(
	    [0] => Error: Invalid MailChimp List ID: 33f4660XXX, code 200
	    [__pid] => 24043
	)

Sure enough, when I check in Mailchimp, the List ID "33f4660XXX" is the one from Account A. However, it should use the one from Account B, as Account B obviously has no access to the subscriber lists in Account A.

So what I think is going on: When submitting a new subscriber via the backend, it correctly uses the API key of the configuration scope/Store View that the customer is assigned to, but it does not use the correct List ID. Instead, it seems to use the List ID from the default configuration.

In the frontend, everything works as expected, so I'm thinking there must be some bug in the extension code.

I hope my error report is understandable enough but feel free to ask if you have any questions. Thanks!

**Comment by mobweb** [comment]
(It seems that I can not re-open this issue, so please re-open it for me, thanks!)

**Comment by centerax** [comment]
Thanks @mobweb We will review this again.

**Comment by Santiagoebizmarts** [comment]
@mobweb Thanks for taking the time to explain all this. You can see the commit https://github.com/ebizmarts/magemonkey/commit/657713ddaff8c624078b81cb547ddca94bcb78e4.
This is based on our latest version 1.1.30.
If you want to install this you can wait until we release the new version or download it from github and then do a manual installation.

**Comment by mobweb** [comment]
Hm, I just cloned the latest version of your repository and uploaded it to our testing server. I still get the same error when I'm trying to subscribe a customer from website B to the linked Mailchimp account/list:

    2015-04-17T18:43:06+00:00 DEBUG (7): Array
    (
        [0] => Error: Invalid MailChimp List ID: 33f46XXX, code 200
        [__pid] => 8389
    )

**Comment by mobweb** [comment]
Can you tell if the problem is on my side or can you reproduce that error?

**Comment by centerax** [comment]
Hello @mobweb 

We will let you know once we work on this issue, no ETA I'm afraid.

Thanks.

**Comment by Santiagoebizmarts** [comment]
Hello @mobweb,

Sorry for the delay on this.
I think I found the problem. When getting the store it seemed to get the default's in stead of the one you really are on.
I made some changes and it should work okay. The commit is https://github.com/ebizmarts/magemonkey/commit/625c5149fffeeaa4a8246dbac1ca72ab6d460c95
If you have any other problem just post it here and I will re open this issue and attend you right away.

Regards.
